pkg/kubelet/kubelet_node_status.go

```diff
@@ -91,7 +93,8 @@ func (kl *Kubelet) tryRegisterWithAPIServer(node *v1.Node) bool {
 		return true
 	}
 
-	if !apierrors.IsAlreadyExists(err) {
+	continueOnForbidden := !utilfeature.DefaultFeatureGate.Enabled(features.DisableKubeletRegisterGetOnForbidden)
+	if !(apierrors.IsAlreadyExists(err) || apierrors.IsForbidden(err) && continueOnForbidden) {
```

Comment:
I think we should split it into two IF statements to simplify its logic

Reply:
I made it a switch instead. PTAL.